sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -236,21 +243,30 @@ public EventHubProducerClient(string connectionString) : this(connectionString,
         ///
         /// <param name="connection">The connection to use as the basis for delegation of client-type operations.</param>
         /// <param name="transportProducer">The transport producer instance to use as the basis for service communication.</param>
+        /// <param name="partitionProducerPool">A <see cref="TransportProducerPool"/> used to manage a set of partition specific <see cref="TransportProducer"/>.</param>
         ///
         /// <remarks>
         ///   This constructor is intended to be used internally for functional
         ///   testing only.
         /// </remarks>
         ///
         internal EventHubProducerClient(EventHubConnection connection,
-                                        TransportProducer transportProducer)
+                                        TransportProducer transportProducer,
+                                        TransportProducerPool partitionProducerPool = default)
         {
             Argument.AssertNotNull(connection, nameof(connection));
             Argument.AssertNotNull(transportProducer, nameof(transportProducer));
 
+            if (partitionProducerPool == default)
+            {
+                partitionProducerPool = new TransportProducerPool();
+            }
+
             OwnsConnection = false;
             Connection = connection;
             EventHubProducer = transportProducer;
+            PartitionProducerPool = partitionProducerPool;
```

Comment:
You may want to consider simplifying to:  `PartitionProducerPool = partitionProducerPool ?? new TransportProducerPool();`